common/cache/lru.go

```diff
@@ -300,20 +317,22 @@ func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool)
 		entry.createTime = c.clock.Now().UTC()
 	}
 
-	if len(c.byKey) >= c.maxSize {
+	if c.currSize >= c.maxSize {
```

Comment:
I remember this is a `for` loop in the previous version

Reply:
I changed it back, from current implementation the LRU will not pop out any item if the reference is not 0. so the for loop here will introduce an infinite loop.